servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/amqp/SendLinkHandler.java

```diff
@@ -71,23 +63,16 @@ public void onDelivery(Event event)
     }
 
     @Override
-    public void onLinkFlow(Event event)
-    {
-        if (this.isFirstFlow)
```

Comment:
[SpotBugs-P2]
Possible doublecheck on com.microsoft.azure.servicebus.amqp.SendLinkHandler.isFirstFlow in com.microsoft.azure.servicebus.amqp.SendLinkHandler.onLinkFlow(Event)

This method may contain an instance of double-checked locking.  This idiom is not correct according to the semantics of the Java memory model.  For more information, see the web page http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html.